### PR TITLE
Update storage

### PR DESCRIPTION
**Pull request by brianyenna**
## Description
The changes made here update the Storage code with minor bug fixes, but focuses mainly on establishing comprehensive tests for each of the major storage classes (namely JsonMentorListStorage, JsonParticipantListStorage, JsonTeamListStorage and AlfredStorageManager). The tests also check that appropriate exceptions are thrown when edge cases (such as invalid values in the .json files) are encountered. 

This PR targets #20 and #21.

## Checks
- [X] No errors when running the tests
- [X] Build and checkstyle passes
- [] Run the app and ran 2 commands without failing
- [X] Written the baseline test cases for the PR

## Changelog
- []
- []
- []

## Comments for Reviewer (if any)
P.s. @hcwong take note that as per our conversation, the SubjectName#isValid() and ProjectType#isValid() methods have been altered to return this.name() instead;

### CODE REVIEW

File: src/test/java/seedu/address/testutil/TypicalTeams.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import seedu.address.model.entity.ProjectType;
+import seedu.address.model.entity.Score;
+import seedu.address.model.entity.SubjectName;
+import seedu.address.model.entity.Team;
+import seedu.address.model.entitylist.TeamList;
+
+/**
+ * A utility class containing a list of {@code Team} objects to be used in tests.
+ */
+public class TypicalTeams {
```
Comment (hcwong) [inline src/test/java/seedu/address/testutil/TypicalTeams.java]: Might have a problem here with static variables. Because let's say we mutate the arttributes of them in 1 test, they will be affected in the next test because both point to the same variable.  Will need a reset test. I wrote one in my PR
